catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/SolrMetacardClientImpl.java

```diff
@@ -110,14 +117,39 @@ public SourceResponse query(QueryRequest request) throws UnsupportedQueryExcepti
 
         SolrQuery query = getSolrQuery(request, filterDelegateFactory.newInstance(resolver));
 
+        boolean isFacetedQuery = false;
+        FacetProperties textFacetsProp = (FacetProperties) request.getPropertyValue(FACET_FIELDS_KEY);
```

Comment:
Should we check if the result of `getPropertyValue` is actually a `FacetProperties` before casting it?